src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size + 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("The index passed is invalid");
+        }
+        if (index == capacity || size == capacity - 1) {
+            elementData = grow();
+        }
```

Comment:
You have a redundant check for `index == capacity`. The check for `size == capacity - 1` is enough to determine if the array needs to grow. Also, consider extracting the growth logic into a separate method as per the checklist requirement for resizing the array in a separate method.